src/gen9_avc_encoder.c

```diff
@@ -3596,6 +3662,456 @@ gen9_avc_set_curbe_mbenc(VADriverContextP ctx,
 }
 
 static void
+gen9_avc_fei_set_curbe_mbenc(VADriverContextP ctx,
+                             struct encode_state *encode_state,
+                             struct i965_gpe_context *gpe_context,
+                             struct intel_encoder_context *encoder_context,
+                             void * param)
+{
+    struct i965_driver_data *i965 = i965_driver_data(ctx);
+    gen9_avc_fei_mbenc_curbe_data *cmd;
+    struct encoder_vme_mfc_context * vme_context = (struct encoder_vme_mfc_context *)encoder_context->vme_context;
+    struct generic_enc_codec_state * generic_state = (struct generic_enc_codec_state *)vme_context->generic_enc_state;
+    struct avc_enc_state * avc_state = (struct avc_enc_state *)vme_context->private_enc_state;
+    VASurfaceID surface_id;
+    struct object_surface *obj_surface;
+    VAEncSliceParameterBufferH264 * slice_param = avc_state->slice_param[0];
+    VAEncPictureParameterBufferH264  *pic_param = avc_state->pic_param;
+    VAEncMiscParameterFEIFrameControlH264 *fei_param = avc_state->fei_framectl_param;
+
+    struct mbenc_param * curbe_param = (struct mbenc_param *)param ;
+    unsigned char qp = 0;
+    unsigned char me_method = 0;
+    unsigned int mbenc_i_frame_dist_in_use = curbe_param->mbenc_i_frame_dist_in_use;
+    unsigned int table_idx = 0;
+    int ref_width, ref_height, len_sp;
+    int is_bframe = (generic_state->frame_type == SLICE_TYPE_B);
+    unsigned int preset = generic_state->preset;
+
+    qp = pic_param->pic_init_qp + slice_param->slice_qp_delta;
+
+    cmd = (gen9_avc_fei_mbenc_curbe_data *)i965_gpe_context_map_curbe(gpe_context);
+    memset(cmd, 0, sizeof(gen9_avc_fei_mbenc_curbe_data));
```

Comment:
Again not much use unless we fix the i965_gpe_context_map_curbe() code to check whether drm_bo mapped successfully or not.